virtualizarr/tests/test_writers/test_icechunk.py

```diff
@@ -285,6 +286,422 @@ def test_write_loadable_variable(
+        icechunk_filestore_append = IcechunkStore.open_existing(
+            storage=icechunk_storage, mode="a"
+        )
+        dataset_to_icechunk(vds, icechunk_filestore_append, append_dim="x")
+        array = open_zarr(icechunk_filestore_append, consolidated=False, zarr_format=3)
+
+        expected_ds = open_dataset(simple_netcdf4)
+        expected_array = concat([expected_ds, expected_ds], dim="x")
+        xrt.assert_identical(array, expected_array)
+
+    ## When appending to a virtual ref with encoding, it succeeds
+    def test_append_virtual_ref_with_encoding(
+        self, icechunk_storage: "StorageConfig", netcdf4_files_factory: Callable
+    ):
+        import xarray.testing as xrt
+        from icechunk import IcechunkStore
+
+        scale_factor = 0.01
+        encoding = {"air": {"scale_factor": scale_factor}}
+        filepath1, filepath2 = netcdf4_files_factory(encoding=encoding)
+        vds1, vds2 = (
+            gen_virtual_dataset(
+                file_uri=filepath1,
+                shape=(1460, 25, 53),
+                chunk_shape=(1460, 25, 53),
+                dims=["time", "lat", "lon"],
+                dtype=np.dtype("float64"),
+                variable_name="air",
+                encoding={"scale_factor": scale_factor},
+                base_offset=15419,
+                length=15476000,
+            ),
+            gen_virtual_dataset(
+                file_uri=filepath2,
+                shape=(1460, 25, 53),
+                chunk_shape=(1460, 25, 53),
+                dims=["time", "lat", "lon"],
+                dtype=np.dtype("float64"),
+                variable_name="air",
+                encoding={"scale_factor": scale_factor},
+                base_offset=15419,
+                length=15476000,
+            ),
+        )
+
+        # create the icechunk store and commit the first virtual dataset
+        icechunk_filestore = IcechunkStore.create(storage=icechunk_storage)
+        dataset_to_icechunk(vds1, icechunk_filestore)
+        icechunk_filestore.commit(
+            "test commit"
+        )  # need to commit it in order to append to it in the next lines
+
+        # Append the same dataset to the same store
+        icechunk_filestore_append = IcechunkStore.open_existing(
+            storage=icechunk_storage, mode="a"
+        )
+        dataset_to_icechunk(vds2, icechunk_filestore_append, append_dim="time")
+        new_ds = open_zarr(icechunk_filestore_append, consolidated=False, zarr_format=3)
+
+        expected_ds1, expected_ds2 = open_dataset(filepath1), open_dataset(filepath2)
+        expected_ds = concat([expected_ds1, expected_ds2], dim="time").drop_vars(
+            ["lon", "lat", "time"], errors="ignore"
+        )
+        # Because we encode attributes, attributes may differ, for example
+        # actual_range for expected_ds.air is array([185.16, 322.1 ], dtype=float32)
+        # but encoded it is [185.16000366210935, 322.1000061035156]
```

Comment:
@TomNicholas I cannot xrt.assert_identical when the encoded value returned from xarray.backend.zarr.encode_zarr_attr_value is different from the actual value. Let me know if you think assert_equal is acceptable or there is another route to try.

Reply:
I think that's okay? Not totally sure though. @mpiannucci do you know?